Unify use of `enableCollectiveHashDebug_` and trivial updates

oraluben opened the pull request:
Use `enableCollectiveHashDebug_` instead of checking env ad-hoc when `TORCH_DISTRIBUTED_DEBUG = DETAIL`

Fixes #ISSUE_NUMBER


cc @H-Huang @awgu @kwen2501 @wanchaol @fegin @fduwjj @wz337 @wconstab @d4l3k @c-p-i-o

kwen2501 reviewed (approved):
Is there a typo or we did name the variable like that earlier?

oraluben committed:
Remove `getIndexFromDeviceKey`

oraluben committed:
Unify use of `enableCollectiveHashDebug_` (and fix typo)

Use `enableCollectiveHashDebug_` instead of checking env ad-hoc when `TORCH_DISTRIBUTED_DEBUG = DETAIL`

fegin reviewed (approved):
@kwen2501 Nice catch!

kwen2501 commented:
@pytorchbot merge

oraluben commented:
@pytorchbot merge

oraluben commented:
@pytorchbot merge

oraluben commented:
@pytorchbot merge

oraluben commented:
@kwen2501 Could you please merge this?

oraluben commented:
ping

oraluben commented:
@pytorchbot rebase

oraluben commented:
@pytorchbot merge

kwen2501 commented:
Sorry I was on PTO and missed your request.

For the lint issues, you can run:
lintrunner -a
to fix them.

Cc @fegin can you please help the contributor in case there are more merge blockers? Thanks!

oraluben commented:
@kwen2501 Thanks! I think the lint is fine, but the CI need to be approved by member to run.

oraluben commented:
@pytorchbot merge

oraluben commented:
@pytorchbot rebase

oraluben commented:
ping, we need another approve to run lint here :)

cyyever commented:
@pytorchbot merge -r

cyyever commented:
@pytorchbot rebase

cyyever commented:
@pytorchbot merge -i
